src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -1584,7 +1584,11 @@ private ConsumeNumberResult ConsumeZero(ref ReadOnlySpan<byte> data, ref int i)
             if (nextByte != '.' && nextByte != 'E' && nextByte != 'e')
             {
                 _bytePositionInLine += i;
-                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, nextByte);
+
+                if (nextByte >= '0' && nextByte <= '9')
```

Comment:
We'd want to make a similar fix in `Utf8JsonReader.MultiSegment.cs`.

https://github.com/dotnet/corefx/blob/9be99cb634ffe1b8dc60e4b3d8c01026347a0283/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L1344-L1348

Reply:
I would instead consider moving the condition to the `ThrowJsonReaderException` itself under the `ExpectedEndOfDigitNotFound` switch case. I believe, the only case where we throw this exception and nextByte is not a digit is this particular one:
https://github.com/dotnet/corefx/blob/9be99cb634ffe1b8dc60e4b3d8c01026347a0283/src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs#L308-L310

Reply:
That way we wouldn't need to make any changes in Utf8JsonReader.MultiSegment.cs as well. Good.